crates/re_renderer/shader/rectangle_fs.wgsl

```diff
@@ -40,53 +40,77 @@ fn decode_color(sampled_value: Vec4) -> Vec4 {
     return rgba;
 }
 
+/// Takes a floating point texel coordinate and outputs a neighrest neighbor
+fn to_clamp_to_border_sample(coord: Vec2, texture_dimension: Vec2) -> IVec2 {
+    return IVec2(clamp(floor(coord), Vec2(0.0), texture_dimension - Vec2(1.0)));
+}
+
+/// Takes a floating point texel coordinate and outputs the four integer texel coordinates that are used for bilinear filtering.
+/// All four samples are clamped to the texture border.
+fn clamped_bilinear_sample_positions(coord: Vec2, texture_dimension: Vec2) -> array<IVec2, 4> {
+    let v00 = to_clamp_to_border_sample(coord + vec2(-0.5, -0.5), texture_dimension);
+    let v01 = to_clamp_to_border_sample(coord + vec2(-0.5,  0.5), texture_dimension);
+    let v10 = to_clamp_to_border_sample(coord + vec2( 0.5, -0.5), texture_dimension);
+    let v11 = to_clamp_to_border_sample(coord + vec2( 0.5,  0.5), texture_dimension);
+    return array<IVec2, 4>(IVec2(v00), IVec2(v01), IVec2(v10), IVec2(v11));
+}
+
 fn filter_bilinear(coord: Vec2, v00: Vec4, v01: Vec4, v10: Vec4, v11: Vec4) -> Vec4 {
-    let top = mix(v00, v10, fract(coord.x - 0.5));
-    let bottom = mix(v01, v11, fract(coord.x - 0.5));
-    return mix(top, bottom, fract(coord.y - 0.5));
+    let top = mix(v00, v10, fract(coord.x - 0.5 + 10.0));
+    let bottom = mix(v01, v11, fract(coord.x - 0.5 + 10.0));
+    return mix(top, bottom, fract(coord.y - 0.5 + 10.0));
```

Comment:
Did you remember to self-review your code before opening a PR?

Reply:
😳